### PR TITLE
feat(api):Developed Cart API

### LINKED ISSUE #19: Cart API

### PR DESCRIPTION
**Pull request by jhoolwerf**

### CODE REVIEW

**Review by MShiers7 — APPROVED** [review]

File: api/cart.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const cartRouter = express.Router;
+const { Cart } = require('../db/models/index');
+
+cartRouter.post("/", async (req, res, next) => {
+    try {
+        const { orderId, quantity } = req.body;
+        const newOrder = await addItemToCart ({ orderId, quantity });
```
Comment (Remics89) [inline api/cart.js]: - Now that you have `{ Cart }` deconstructed on Line 3, youll want to call the db method as an object key/value:

```suggestion
        const newOrder = await Cart.addItemToCart ({ orderId, quantity });
```

File: api/cart.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        throw error;
+    }
+})
+
+cartRouter.delete("/:cartId", async (req, res, next) => {
+    try {
+        const { cartId } = req.params;
+        const cart = await getOrderById(id);
+        if (req.cart === cart.cartId) {
+            const deletedCart = await destroyCart(cartId);
```
Comment (Remics89) [inline api/cart.js]: - The same for here and for line 19.
- For line 19, be careful of the fact that getOrderById is a db method for Orders, not Cart.
```suggestion
            const deletedCart = await Cart.destroyCart(cartId);
```

File: api/cart.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        throw error;
+    }
+})
+
+cartRouter.patch("/:cartId", async (req, res, next) => {
+    try {
+        const { cartId } = req.params;
+        const cart = await getOrderById(id);
+        if (req.cart === cart.cartId) {
+            const updatedCart = await updateCart(cartId);
```
Comment (Remics89) [inline api/cart.js:36]: - This one as well as line 34
- For line 34, be careful of the fact that getOrderById is a db method for Orders, not Cart.

```suggestion
            const updatedCart = await Cart.updateCart(cartId);
```

File: api/index.js [hunk 4]
```diff
@@ -14,4 +14,8 @@ apiRouter.get('/health', (req, res, next) => {
 
 // place your routers here
 
+//ROUTER: /api/cart
+const cartRouter = require('./cart');
+Router.use("/cart", cartRouter);
```
Comment (Remics89) [inline api/index.js:19]: Youll want to reference apiRouter here.

```suggestion
apiRouter.use("/cart", cartRouter);
```

**Review by Remics89 — CHANGES_REQUESTED** [review]

**Review by Remics89 — APPROVED** [review]

**Pull request merged**